src/CBT.Core/Internal/NuGetPackageReferenceProjectParser.cs

```diff
@@ -17,33 +19,55 @@ public NuGetPackageReferenceProjectParser(CBTTaskLogHelper logger)
+                _log.LogWarning($"Missing expected assests file directory.  This is typically because the flag generated at $(CBTModuleNuGetAssetsFlagFile) does not exist or is empty.  Ensure the GenerateModuleAssetFlagFile target is running. It may also be because the CBTModules.proj does not import cbt build.props in some fashion.");
                 yield break;
             }
             VersionFolderPathResolver versionFolderPathResolver = new VersionFolderPathResolver(packagesPath);
-            
-            string lockFilePath = Path.Combine(assetsFileDirectory, LockFileFormat.AssetsFileName);
+
+            string lockFilePath = Path.Combine(packageRestoreData.RestoreOutputAbsolutePath, LockFileFormat.AssetsFileName);
 
             if (!File.Exists(lockFilePath))
             {
                 _log.LogWarning($"Missing expected nuget assests file '{lockFilePath}'.  If you are redefining BaseIntermediateOutputPath ensure it is unique per project. ");
                 yield break;
             }
+            HashSet<string> processedPackages = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
             LockFile lockFile = LockFileUtilities.GetLockFile(lockFilePath, NullLogger.Instance);
 
-            foreach (LockFileLibrary library in lockFile.Libraries)
+            foreach (var pkg in packageRestoreData.PackageImportOrder)
             {
-                yield return new PackageIdentityWithPath(library.Name, library.Version, versionFolderPathResolver.GetPackageDirectory(library.Name, library.Version), versionFolderPathResolver.GetInstallPath(library.Name, library.Version));
+                var dependencies = lockFile.Targets.First().Libraries
+                    .Where(lib => lib.Name.Equals(pkg.Id, StringComparison.OrdinalIgnoreCase) &&
+                                  lib.Version.ToString().Equals(pkg.Version, StringComparison.OrdinalIgnoreCase)).Select(lib => lib.Dependencies).SelectMany(p => p.Select(i => i));
+                foreach (PackageDependency dependency in dependencies)
+                {
+                    // In the <PackageReference scenario nuget will only install one packageId.  If you have two packages that reference different package versions of a third package then it will choose the common highest version and if there is no common version it will error.  If you have two packages listed with two different versions it will choose the first entry and silently not install the other.
+                    // If the package is already processed then skip.  If the package is explicitly added then skip to use that order.
+                    if (!processedPackages.Contains($"{dependency.Id}") && !packageRestoreData.PackageImportOrder.Any(pio => pio.Id.Equals($"{dependency.Id}", StringComparison.OrdinalIgnoreCase)))
```

Comment:
`$"{dependency.Id}"` looks like it was left over from some previous logic, just make it `dependency.Id`